pytensor/link/pytorch/dispatch/basic.py

```diff
@@ -132,3 +134,28 @@ def makevector(*x):
         return torch.tensor(x, dtype=torch_dtype)
 
     return makevector
+
+
+@pytorch_funcify.register(OpFromGraph)
+def pytorch_funcify_OpFromGraph(op, node=None, **kwargs):
+    _ = kwargs.pop("storage_map", None)
+
+    fgraph_fn = torch.compile(pytorch_funcify(op.fgraph, **kwargs))
+
+    if len(op.fgraph.outputs) > 1:
+
+        def inner(*args):
+            return fgraph_fn(*args)
+    else:
+
+        def inner(*args):
+            return fgraph_fn(*args)[0]
+
+    # Don't compile the inner function
+    # This is due torch failing to create
+    # guards when parent scoped closure variables
+    # are used in conditional statements.
+    # Instead of rewriting many portions of code
+    # this will allow for only this small section to
+    # not be compiled by the outer graph
+    return torch.compiler.disable(inner)
```

Comment:
Is this the two inner functions? Would it not be needed if the if else is inside `inner`?

Reply:
Has something bizarre to do with the combination of `fgraph_fn` being a bunch of nested functions, and this inner function being nested. The bigger part of that torch compiler isn't super great at handling conditionals user closure variables, at least in pytensor. It would probably need a much deeper dive. It looks like it might be something that can happen with other functions.

Reply:
That's worrisome. What error did you get without this disabling?

Reply:
I have the error above in a comment, but it's essentially going to say the generated code from pytensor can't find some functions (all the inner functions returned in torch dispatch)